app/build.gradle.kts

```diff
@@ -2,6 +2,16 @@ plugins {
     id("com.android.application")
     id("org.jetbrains.kotlin.android")
     id("androidx.navigation.safeargs.kotlin")
+    id("io.gitlab.arturbosch.detekt")
+}
+
+detekt {
+    // Version of detekt that will be used. When unspecified the latest detekt
+    // version found will be used. Override to stay on the same version.
+    toolVersion = "1.23.3"
+    buildUponDefaultConfig = true // preconfigure defaults
+    allRules = false // activate all available (even unstable) rules.
+    config.setFrom("$rootDir/detekt.yml") // point to your custom config defining rules to run, overwriting default behavior
```

Comment:
v momente kdy pouzijes
buildUponDefaultConfig = true
neni potreba v tom detekt.yml definovat behavior pro vsechno, jak pises v komentari staci jen to co potrebujes overridovat

Reply:
jo to mi tady zbyly komentáře z guidy :( 

Byl bych pro následující:

- `detekt.yml` obsahuje všechna známá nastavení (vygenerováno pomocí gradle tasku)
- změna verze `toolVersion` tak nezmění chování linteru
- při přechodu na vyšší verzi nebo přidáním pravidel přegenerovat základní nastavení pomocí gradle tasku (je potřeba smazat/přejmenovat `detekt.yml` a zkontrolovat co se změnilo)

Nebo, jak to máte na projektech do teď?

Reply:
doted to na projektech nemame, ale tvoje vysvetleni dava smysl. Jak se potom resi kdyz si nejaka ta nastaveni upravime a pak prijde nova verze? 
- vygeneruje se novy soubor a manualne se to projde 
- nebo jde specifikovat nejaky detekt-quanti.yml, ktery by jeste overridoval ten detekt.yml?